components/gitpod-db/src/typeorm/user-db-impl.ts

```diff
@@ -47,17 +47,24 @@ import { DBWorkspace } from "./entity/db-workspace";
 import { DBUserSshPublicKey } from "./entity/db-user-ssh-public-key";
 import { TypeORM } from "./typeorm";
 import { log } from "@gitpod/gitpod-protocol/lib/util/logging";
+import { DataCache } from "../data-cache";
 
 // OAuth token expiry
 const tokenExpiryInFuture = new DateInterval("7d");
 
 /** HACK ahead: Some entities - namely DBTokenEntry for now - need access to an EncryptionService so we publish it here */
 export let encryptionService: EncryptionService;
 
+const userCacheKeyPrefix = "user:";
+function getUserCacheKey(id: string): string {
+    return userCacheKeyPrefix + id;
+}
```

Comment:
The key is specific to a particular type of implementation (redis), but our service only depends on a generic interface. This feels wrong. 

What can we do here?

Secondly, it would be nice to have the keys and the namespaces defined in a single place to easily ensure we dont' clash. This living next to the Redis implementation seems most sensible

Reply:
They are adhering to the contract specified in the DataCache interface. Redis happens to fulfill that contract naturally 😇

Reply:
Secondly, it would be nice to have the keys and the namespaces defined in a single place to easily ensure we dont' clash. This living next to the Redis implementation seems most sensible

Reply:
I disagree, because as you said it is not redis specific so it does not belong to that implementation. The space for names is an instance of a cache. We could add the prefixes to the data-cache.ts file. I prefer it on the caller side tbh, though.

Reply:
Fair, I only worry that we're making it way too easy to clash on the keyspace

Reply:
A solution would be to use a second db in redis. You can have multiple "instances" inside a given redis deployment, which are effectively isolated from each other. Right now, we're just using the default (db:0) but we could point to db:1 for the domain object cache